homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,244 @@
+                if "No route to host" in str(err):
+                    return False, None, "stream_no_route_to_host"
+                if "Input/output error" in str(err):
+                    return False, None, "stream_io_error"
+                raise err
+        return True, fmt, ""
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
+            vol.Optional(
+                CONF_STILL_IMAGE_URL,
+                description={
+                    "suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")
+                },
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+            ): str,
+            vol.Optional(
+                CONF_PASSWORD,
+                description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+            ): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+            ): bool,
+            vol.Optional(
+                CONF_FRAMERATE,
+                description={"suggested_value": user_input.get(CONF_FRAMERATE, "")},
+            ): int,
+            vol.Optional(
+                CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL)
+            ): bool,
+        }
+        return vol.Schema(spec)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the start of the config flow."""
+        errors = {}
+        if user_input:
+            # Secondary validation because serialised vol can't seem to handle this complexity:
+            if not user_input.get(CONF_STILL_IMAGE_URL) and not user_input.get(
+                CONF_STREAM_SOURCE
+            ):
+                errors["base"] = "no_still_image_or_stream_url"
+            else:
+                (res, still_format, errors["base"]) = await self._test_connection(
+                    user_input
+                )
+                if res:
+                    user_input[CONF_CONTENT_TYPE] = still_format
+                    return self.async_create_entry(
+                        title=user_input[CONF_NAME], data=user_input
```

Comment:
Almost all (or all of these) can be changed later so we want to store them in `options` instead of `data`

See https://github.com/home-assistant/core/blob/dev/homeassistant/components/mjpeg/config_flow.py#L159

Reply:
Yes, makes sense. I'm making pretty good progress on it. Will push once it's all working with options flow.  MJPEG is a great reference and have been using that.

Reply:
Settings are now in options flow. Done.